src/reduced_order/rom_test_location.cpp

```diff
@@ -11,41 +11,65 @@
+ROMTestLocation<dim, nstate>::ROMTestLocation(const RowVectorXd& parameter, std::unique_ptr<ROMSolution<dim, nstate>> rom_solution)
         : parameter(parameter)
-        , rom_solution(rom_solution)
+        , rom_solution(std::move(rom_solution))
         , mpi_communicator(MPI_COMM_WORLD)
         , mpi_rank(dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD))
         , pcout(std::cout, mpi_rank==0)
 {
     pcout << "Creating ROM test location..." << std::endl;
     compute_FOM_to_initial_ROM_error();
     initial_rom_to_final_rom_error = 0;
-    compute_total_error();
+    total_error = fom_to_initial_rom_error;
+
     pcout << "ROM test location created. Error estimate updated." << std::endl;
 }
 
 template <int dim, int nstate>
 void ROMTestLocation<dim, nstate>::compute_FOM_to_initial_ROM_error(){
     pcout << "Computing adjoint-based error estimate between ROM and FOM..." << std::endl;
-    dealii::LinearAlgebra::distributed::Vector<double> gradient(rom_solution->right_hand_side);
-    dealii::LinearAlgebra::distributed::Vector<double> adjoint(rom_solution->right_hand_side);
 
-    gradient = rom_solution->gradient;
+    dealii::ParameterHandler dummy_handler;
+    std::unique_ptr<FlowSolver::FlowSolver<dim,nstate>> flow_solver = FlowSolver::FlowSolverFactory<dim,nstate>::select_flow_case(&rom_solution->params, dummy_handler);
+    flow_solver->dg->solution = rom_solution->solution;
+    const bool compute_dRdW = true;
+    flow_solver->dg->assemble_residual(compute_dRdW);
+    dealii::TrilinosWrappers::SparseMatrix system_matrix_transpose = dealii::TrilinosWrappers::SparseMatrix();
+    system_matrix_transpose.copy_from(flow_solver->dg->system_matrix_transpose);
+
+    // Initialize with same sparsity pattern as dg->right_hand_side
+    dealii::LinearAlgebra::distributed::Vector<double> adjoint(flow_solver->dg->right_hand_side);
+
+    dealii::LinearAlgebra::distributed::Vector<double> gradient(rom_solution->gradient);
 
     Parameters::LinearSolverParam linear_solver_param;
-    linear_solver_param.linear_solver_type = Parameters::LinearSolverParam::direct;
-    solve_linear(*rom_solution->system_matrix_transpose, gradient*=-1.0, adjoint, linear_solver_param);
+
+    linear_solver_param.max_iterations = 1000;
+    linear_solver_param.restart_number = 200;
+    linear_solver_param.linear_residual = 1e-17;
+    linear_solver_param.ilut_fill = 50;
+    linear_solver_param.ilut_drop = 1e-8;
+    linear_solver_param.ilut_atol = 1e-5;
+    linear_solver_param.ilut_rtol = 1.0+1e-2;
+    //linear_solver_param.linear_solver_output = Parameters::OutputEnum::verbose;
+    linear_solver_param.linear_solver_type = Parameters::LinearSolverParam::LinearSolverEnum::gmres;
+
+    //linear_solver_param.linear_solver_type = Parameters::LinearSolverParam::direct;
+    solve_linear(system_matrix_transpose, gradient*=-1.0, adjoint, linear_solver_param);
```

Comment:
Can dg's `linear_solver_param` be used here with these values in the parameters file?